scipy/stats/stats.py

```diff
@@ -7469,6 +7470,214 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
+       vol. 7, no. 3, pp. 122-128, 1936,
+       Accessed: Sep. 18, 2019. [Online].
+       Available: https://www.jstor.org/stable/2957563.
+    .. [2] H. A. David and H. N. Nagaraja, "Order Statistics in
+       Nonparametric Inference" in Order Statistics,
+       John Wiley & Sons, Ltd, 2005, pp. 159-170.
+
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.stats import confint_quantile
+    >>> x = np.arange(1, 10)
+    >>> confint_quantile(x, 0.5, 0.95)
+    (2, 8)
+
+    To compute a two-sided interval instead, use the `type` parameter.
+
+    >>> confint_quantile(x, 0.5, 0.99, type='two-sided')
+    (1, 9)
+
+    You can also pass the number of samples as argument (instead of the samples)
+    themselves. The returned values are then the indexes of the upper and lower
+    bounds for the confidence intervals.
+
+    >>> N = 20
+    >>> confint_quantile(N, 0.75, 0.90)
+    (11, 17)
+
+
+    .. versionadded:: 1.6.0
+    """
+
+    ##
+    # Checking the inputs
+    #
+    # x can be either an integer or a one-dimentional array-like
+    if isinstance(x, int):
+        if x < 1:
+            raise ValueError("Invalid parameter: "+repr(x)+", `x` must be either a strictly positive integer or one-dimensional array-like.")
+        n = x
+        return_index = True  # The function will returns the confint indexes
+    else:
+        x = np.asarray(x)
+        if x.ndim != 1:
+            raise ValueError("Invalid parameter: "+repr(x)+", `x` must be either a strictly positive integer or one-dimensional array-like.")
+        x = np.sort(x, axis=0)
+        n = x.shape[0]
+        return_index = False  # The function will returns the confint as values of x
+    #
+    # `confidence` and `quantile` must be between 0 and 1
+    if confidence >= 1 or confidence <= 0:
+        raise ValueError("Invalid `confidence`: "+repr(confidence)+". Provide a real number strictly between 0 and 1.")
+    if quantile >= 1 or quantile <= 0:
+        raise ValueError("Invalid `quantile`: "+repr(quantile)+". Provide a real number strictly between 0 and 1.")
+    #
+    # `type` can be only `one-sided` or `two-sided`
+    if not (type == 'one-sided' or type == 'two-sided'):
+        raise ValueError("Invalid parameter: "+repr(type)+". Valid 'type' values: 'one-sided' or 'two-sided'")
+    ##
+
+    # Handle the type of intervals (one- or two-sided)
+    if type == 'two-sided':
+        conf_working = (1+confidence)/2
+    else:
+        # type == 'one-sided'
+        conf_working = confidence
+
+    # Compute the lower bound
+    LB = _confint_lowerbound(n, quantile, conf_working)
+
+    # Compute the upper bound
+    # -> deduced from the lower bound of (1-quantile)
+    lb = _confint_lowerbound(n, 1-quantile, conf_working)
+    if lb is None:
+        UB = None
+    else:
+        UB = ((n-1) - lb)   # First index is 0 (not 1), hence the -1
+
+    if return_index:
+        return LB, UB
```

Comment:
the docstring doesn't cover how to interpret return of None

Reply:
Fixed with commit https://github.com/scipy/scipy/pull/12680/commits/2a5c2860b168fba87699bc33bcc6238e30420b26